tests/pos/i1747.scala

```diff
@@ -0,0 +1,3 @@
+abstract class Coll[E] extends java.util.Collection[E] {
+  def toArray[T](a: Array[T]): Array[T] = ???
```

Comment:
I don't think that its sound to treat them as equivalent. You can't safely provide a generic Scala Array to Java:

```scala
scala> def foo[T](al: java.util.ArrayList[T], a: Array[T]) = al.toArray(a)
<console>:12: error: type mismatch;
 found   : Array[T]
 required: Array[? with Object]
Note: T >: ? with Object, but class Array is invariant in type T.
You may wish to investigate a wildcard type such as `_ >: ? with Object`. (SLS 3.2.10)
       def foo[T](al: java.util.ArrayList[T], a: Array[T]) = al.toArray(a)
                                                                        ^
```

You need to bound things to:

```scala
scala> def foo[T <: AnyRef](al: java.util.ArrayList[T], a: Array[T]) = al.toArray(a)
foo: [T <: AnyRef](al: java.util.ArrayList[T], a: Array[T])Array[T with Object]
```

The overriding method can't add a bound to the type parameter, but can express the restriction as:

```scala
abstract class C[E] extends java.util.Collection[E] {
   override def toArray[T](a: Array[T with Object]): Array[T with Object] = ???
}
```

We also can't add bridges to paper over the difference because of the mutability of arrays.